stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/viewholder/BaseChannelListItemViewHolder.kt

```diff
@@ -1,10 +1,101 @@
 package io.getstream.chat.android.ui.channel.list.adapter.viewholder
 
+import android.annotation.SuppressLint
+import android.view.MotionEvent
 import android.view.View
 import androidx.recyclerview.widget.RecyclerView
 import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.ui.channel.list.ChannelListView
 import io.getstream.chat.android.ui.channel.list.adapter.diff.ChannelDiff
+import kotlin.math.absoluteValue
 
 public abstract class BaseChannelListItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
     public abstract fun bind(channel: Channel, diff: ChannelDiff)
+
+    @SuppressLint("ClickableViewAccessibility")
+    public fun setSwipeListener(view: View, swipeListener: ChannelListView.SwipeListener) {
```

Comment:
Does this have to be in the base class? It doesn't make sense for `LoadingMoreViewHolder`, which is the only other implementation of this base class.